IngilevichVarvara/cafeMenu.js

```diff
@@ -0,0 +1,250 @@
+//Menu items
+
+//Hamburgers
+//Hamburger types of size
+Hamburger.SIZE_SMALL = {name: "SMALL", price: 50, calories: 20, type: "hamburger"};
+Hamburger.SIZE_LARGE = {name: "SMALL", price: 100, calories: 40, type: "hamburger"};
+
+//Hamburger types of stuffing
+Hamburger.STUFFING_CHEESE = {name: "CHEESE", price: 10, calories: 20, type: "stuffing"};
+Hamburger.STUFFING_SALAD = {name: "SALAD", price: 20, calories: 5, type: "stuffing"};
+Hamburger.STUFFING_POTATO = {name: "POTATO", price: 15, calories: 10, type: "stuffing"};
+
+//Salads
+Salad.CESAR = {name: "CESAR", price: 100, calories: 20};
+Salad.OLIVIER = {name: "OLIVIER", price: 50, calories: 80};
+
+// Drinks
+Drink.COLA = {name: "COLA", price: 50, calories: 40};
+Drink.COFFEE = {name: "COFFEE", price: 80, calories: 20};
+
+
+function MenuItem(price, calories) {
+    this.price = price;
+    this.calories = calories;
+}
+
+/**
+ * Узнать цену гамбургера
+ * @return {Number} Цена в тугриках
+ */
+MenuItem.prototype.calculatePrice = function () {
+    return this.price;
+};
+
+/**
+ * Узнать калорийность
+ * @return {Number} Калорийность в калориях
+ */
+MenuItem.prototype.calculateCalories = function () {
+    return this.calories;
+};
+
+
+//Hamburger
+function Hamburger(size, stuffing) {
+    if (arguments.length > 2) {
+        throw new Error("You have provided too much information about the hamburger. Please specify size and stuffing only.");
+    } else if (arguments.length === 0) {
+        throw new Error("You have not specified size and stuffing of hamburger.");
+    } else if (arguments.length === 1) {
+        let provided_info = Array.prototype.slice.call(arguments)[0];
```

Comment:
Couldn't we use `const` here? We never reassign `provided_info`.

Reply:
There are three usual ways to turn `arguments` object into casual array:

```
var normalArray = Array.prototype.slice.call(arguments);
  // -- or --
  var normalArray = [].slice.call(arguments);
  // -- or --
  var normalArray = Array.from(arguments);
```

Personally, I prefer the last one as it's most brief and concise. It just explicitly tells us what's happening here. Learn more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters#From_arguments_to_an_array